libs/feature/catalog/src/lib/records/records.service.ts

```diff
@@ -1,18 +1,17 @@
 import { Injectable } from '@angular/core'
-import { Observable, of } from 'rxjs'
+import { Observable, of, switchMap } from 'rxjs'
 import { catchError, shareReplay } from 'rxjs/operators'
 import { RecordsRepositoryInterface } from '@geonetwork-ui/common/domain/repository/records-repository.interface'
 
 @Injectable({
   providedIn: 'root',
 })
 export class RecordsService {
```

Comment:
Honestly I think this service is useless and the RecordsRepository can be used directly